tests/protocols/test_http2.py

```
@@ -0,0 +1,1037 @@
+    config.load()
+    assert config.http_protocol_class is ZttpProtocol
+
+
+async def test_config_http_zttp2_loads_http2_protocol():
+    config = Config(app=Response("ok"), http="zttp2")
+    config.load()
+    assert config.http_protocol_class is ZttpH2Protocol
+
+
+class CustomH2Protocol(asyncio.Protocol):
+    alpn_protocols = ["h2", "http/1.1"]
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)